test/unit/model/helpers/downloader-test.js

```diff
@@ -271,4 +258,40 @@ describe('Downloader', function() {
       expect(successHandler).to.be.calledOnce;
     });
   });
+
+  describe('restartDownload', function(){
+    let options = 'http://example.com/jdk.zip';
+
+    it('should change downloader status from \'Download Failed\' to \'Downloading\'', function(){
+      let spy = sandbox.spy(fakeProgress, 'setStatus');
+      downloader.restartDownload();
+      expect(spy).to.have.been.calledOnce;
+      expect(downloader.downloadSize).to.be.equal(0);
+      expect(downloader.received ).to.be.equal(0);
+      expect(downloader.currentSize  ).to.be.equal(0);
+      expect(fakeProgress.setStatus).to.have.been.calledOnce;
+      expect(fakeProgress.setStatus).to.have.been.calledWith('Downloading');
+    });
+
+    it('should call authDownload for entries that requires authentication', function(){
+      let requestGetSpy = sandbox.spy(request, 'get');
+      downloader.setWriteStream(new PassThrough());
+      downloader.restartDownload();
+      downloader.downloadAuth(options, 'username', 'password', 'key', 'sha');
```

Comment:
You are almost there :)
You need to add a little more initialization code before calling `restartDownload()`
``` JavaScript
  let response = new Readable();
  sandbox.stub(request, 'get').returns(response);
  let error = new Error('something bad happened');
  let stream = new Writable();
  downloader.setWriteStream(stream);
  downloader.downloadAuth(options);
  response.emit('error', error);
  // after this downloader will have one failed download in downloads map
  sandbox.stub(downloader, 'downloadAuth');  
  downloader.restartDownload();
  expect(downloader.downloadAuth).to.be.calledOnce;
  expect(downloader.downloadAuth).to.be.calledWith(options, 'username', 'password', 'key', 'sha');
```